### PR TITLE
Added test for strange exception swallowing origamitower/folktale#163

### PR DESCRIPTION
**Pull request by jmatsushita**

### CODE REVIEW

File: test/source/specs/base/concurrency/task.js [hunk 1]
```diff
@@ -539,6 +539,36 @@ describe('Data.Task', () => {
...
+        let Result = undefined;
+        try {
+          let ex = await Task.of('a')
+            .map((() => false ? Result.a : Result.b))
+            .run()
+            .promise();
+          // Should have thrown by then
+          $ASSERT(false)
+        } catch (e) {
+          if (e.message != 'false') $ASSERT(e instanceof TypeError)
```
Comment (robotlolita) [inline test/source/specs/base/concurrency/task.js]: This should be testing if the error is an instance of AssertionError, so:

```js
const { AssertionError } = require('assert');
if (!(e instanceof AssertionError)) {
  $ASSERT(e instanceof TypeError);
}
```

Comment (jmatsushita) [reply test/source/specs/base/concurrency/task.js]: Hi! I tried to use `assert.throws()` which would have been clearer, but it wouldn't compile. I  simplified the test though now. Hopefully it is clearer.

File: test/source/specs/base/concurrency/task.js [hunk 2]
```diff
@@ -539,6 +538,37 @@ describe('Data.Task', () => {
...
+      });
+
+      it('Deals with exception corner cases', async () => {
+        let Result = require('folktale').validation;
+        try {
+          let ex = await Task.of('a')
+            .map((() => false ? Result.a : Result.b))
+            .run()
+            .promise();
+          // Should have thrown a TypeError by then
```
Comment (robotlolita) [inline test/source/specs/base/concurrency/task.js]: Why is this one expected to throw a TypeError here?
